core/domain/state_domain.py

```diff
@@ -1518,25 +1518,25 @@ def update_interaction_customization_args(self, customization_args):
         """
         self.interaction.customization_args = customization_args
 
-    def update_interaction_answer_groups(self, answer_groups_list):
+    def update_interaction_answer_groups(self, answer_group_object):
```

Comment:
Pass a list(AnswerGroup) over here and update the test files accordingly. Once this is fixed I think we can merge this PR! :)

Reply:
But according to the issue we must send an object as a parameter @DubeySandeep https://github.com/oppia/oppia/issues/6271

Reply:
@puneeth2001 Yeah, earlier we were passing a list of dicts and now we will pass a list of answerGroup.

In general, in the update method, we will pass an item which should get updated, like in our case its update_interaction_ _answer_groups_ and asnwer_groups itself is a list(AnswerGroup). Is that helpful? :)